### PR TITLE
Use new image API to save/show image of an entry

### PR DESCRIPTION
**Pull request by AdityaKhatri**
Depends on https://github.com/the-deep/server/pull/669

Depends on https://github.com/the-deep/server/pull/652

### CODE REVIEW

File: src/components/viewer/ImagesGrid/index.js [hunk 1]
```diff
@@ -9,6 +10,36 @@ import ModalBody from '#rscv/Modal/Body';
...
+            alt={imageId}
+            src={source}
+            role="presentation"
+            draggable
+            className={styles.image}
+            onClick={handleImageClick}
+            onDragStart={handleDragStart}
+        />
+    );
+}
```
Comment (tnagorra) [inline src/components/viewer/ImagesGrid/index.js:41]: Image should either have a propTypes definition or should be written in TS

File: src/entities/editEntries.js [hunk 2]
```diff
@@ -165,9 +166,15 @@ export const createEntry = ({
...
 }) => {
+    const {
+        imageDetails,
+    } = data;
+
     const keysToPick = [
         'excerpt',
         'image',
+        'imageRaw',
+        'leadImage',
```
Comment (tnagorra) [inline src/entities/editEntries.js:178]: This seems weird.
1. What is leadImage on entries? Where is this used?
2. What is the difference between image and imageRaw? Do we need both?

File: src/schema/entries.js [hunk 3]
```diff
@@ -33,7 +33,7 @@ const entrySchema = [];
         fields: {
             analysisFramework: { type: 'uint', required: true },
 
-            image: { type: 'string' },
+            image: { type: 'number' },
```
Comment (tnagorra) [inline src/schema/entries.js:52]: Shouldn't we add imageRaw imageDetails here?

File: src/views/EditEntries/Overview/LeftPane/EntriesList/EntryItem/index.js [hunk 4]
```diff
@@ -92,6 +94,8 @@ const EntryItem = (props) => {
                 <EntryPreview
                     entryType={entryType}
                     image={image}
+                    imageRaw={imageRaw}
```
Comment (tnagorra) [inline src/views/EditEntries/Overview/LeftPane/EntriesList/EntryItem/index.js:95]: Do we even need to pass `image` here?

File: src/views/EditEntries/Overview/LeftPane/EntriesList/index.js [hunk 5]
```diff
@@ -74,6 +77,8 @@ export default class EntriesList extends React.PureComponent {
             entryServerId: currentEntryServerId,
             entryType,
             image,
+            imageRaw,
```
Comment (tnagorra) [inline src/views/EditEntries/Overview/LeftPane/EntriesList/index.js:80]: Check if image should be passed

File: src/views/EditEntries/Overview/LeftPane/EntriesList/index.js [hunk 6]
```diff
@@ -54,8 +54,11 @@ export default class EntriesList extends React.PureComponent {
 
         const entryData = entryAccessor.data(entry);
 
+        const imageDetails = entryAccessor.imageDetails(entry);
+
         const {
             entryType,
+            imageRaw,
```
Comment (tnagorra) [inline src/views/EditEntries/Overview/LeftPane/EntriesList/index.js:61]: Check if image should be passed

File: src/views/EditEntries/Overview/index.js [hunk 7]
```diff
@@ -194,7 +194,13 @@ export default class Overview extends React.PureComponent {
...
         } = this.props;
 
-        const { type, value, dropped } = excerptData;
+        const {
+            type,
+            value,
+            dropped,
+            imageDetails,
+        } = excerptData;
+        console.warn('i am here', imageDetails);
```
Comment (tnagorra) [inline src/views/EditEntries/Overview/index.js]: No console.warn for debugging.

File: src/views/EditEntries/WidgetFaram/index.js [hunk 8]
```diff
@@ -446,10 +452,12 @@ export default class WidgetFaram extends React.PureComponent {
             excerpt,
             droppedExcerpt,
             image,
```
Comment (tnagorra) [inline src/views/EditEntries/WidgetFaram/index.js]: Check if we need to pass image here

File: src/views/EditEntries/WidgetFaram/index.js [hunk 9]
```diff
@@ -466,6 +474,8 @@ export default class WidgetFaram extends React.PureComponent {
                 excerpt,
                 droppedExcerpt,
                 image,
```
Comment (tnagorra) [inline src/views/EditEntries/WidgetFaram/index.js]: Check if we need to pass image here

File: src/widgets/tagging/Excerpt/index.js [hunk 10]
```diff
@@ -186,12 +189,16 @@ export default class Excerpt extends React.PureComponent {
     }
 
     renderExcerptImage = () => {
-        const { image } = this.props;
+        const {
+            image,
```
Comment (tnagorra) [inline src/widgets/tagging/Excerpt/index.js]: Do we even need this?

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: src/components/general/Navbar/NavDrop/index.js [hunk 11]
```diff
@@ -103,8 +103,8 @@ DropItem.defaultProps = {
 const mapStateToProps = state => ({
     activeProjectId: activeProjectIdFromStateSelector(state),
     activeCountryId: activeCountryIdFromStateSelector(state),
-    activeUser: activeUserSelector(state),
-    userInformation: currentUserInformationSelector(state),
+    activeUser: activeUserSelector(state), // from auth
+    userInformation: currentUserInformationSelector(state), // from users
```
Comment (tnagorra) [inline src/components/general/Navbar/NavDrop/index.js]: No need to keep this comment

File: src/components/viewer/Attachment/index.js [hunk 12]
```diff
@@ -91,7 +94,7 @@ export default class Attachment extends React.PureComponent {
 
                     return (
                         <InternalGallery
-                            galleryId={attachment && attachment.id}
+                            attachmetn={attachment}
```
Comment (tnagorra) [inline src/components/viewer/Attachment/index.js]:
```suggestion
                            attachment={attachment}
```

**Review by samshara — APPROVED** [review]

File: src/entities/editEntries.js (hunk 2 above)
Comment (AdityaKhatri) [reply src/entities/editEntries.js:178]: 1. LeadImage is the `id` of image coming from lead if we are dragging from the `images column.
2. ImageRaw is the blob we save before creating of image file
3. Image is the generated file URL

File: src/views/EditEntries/Overview/LeftPane/EntriesList/EntryItem/index.js (hunk 4 above)
Comment (AdityaKhatri) [reply src/views/EditEntries/Overview/LeftPane/EntriesList/EntryItem/index.js:95]: No, I'll remove it

**Review by samshara — APPROVED** [review]

File: src/schema/entries.js (hunk 3 above)
Comment (tnagorra) [reply src/schema/entries.js:52]: ...

File: src/views/EditEntries/Overview/LeftPane/EntriesList/EntryItem/index.js (hunk 4 above)
Comment (tnagorra) [reply src/views/EditEntries/Overview/LeftPane/EntriesList/EntryItem/index.js:95]: When?

**Review by tnagorra — APPROVED** [review]

**Review by tnagorra — APPROVED** [review]

File: src/views/EditEntries/Overview/LeftPane/EntriesList/index.js (hunk 5 above)
Comment (tnagorra) [reply src/views/EditEntries/Overview/LeftPane/EntriesList/index.js:80]: ...

File: src/views/EditEntries/Overview/LeftPane/EntriesList/index.js (hunk 6 above)
Comment (tnagorra) [reply src/views/EditEntries/Overview/LeftPane/EntriesList/index.js:61]: ...

File: src/components/general/Navbar/NavDrop/index.js (hunk 11 above)
Comment (tnagorra) [reply src/components/general/Navbar/NavDrop/index.js]: ...

File: src/views/EditEntries/Overview/LeftPane/EntriesList/index.js (hunk 5 above)
Comment (AdityaKhatri) [reply src/views/EditEntries/Overview/LeftPane/EntriesList/index.js:80]: It is not requred

File: src/views/EditEntries/Overview/LeftPane/EntriesList/index.js (hunk 6 above)
Comment (AdityaKhatri) [reply src/views/EditEntries/Overview/LeftPane/EntriesList/index.js:61]: not required

**Review by tnagorra — APPROVED** [review]

**Review by tnagorra — APPROVED** [review]

**Pull request merged**